types/index.d.ts

```diff
@@ -4248,14 +4383,26 @@ declare namespace math {
      * Calculate the absolute value of a number. For matrices, the function
      * is evaluated element wise.
      */
-    abs(): MathJsChain
+    abs(this: MathJsChain<number>): MathJsChain<number>
+    abs(this: MathJsChain<BigNumber>): MathJsChain<BigNumber>
+    abs(this: MathJsChain<Fraction>): MathJsChain<Fraction>
+    abs(this: MathJsChain<Complex>): MathJsChain<Complex>
+    abs(this: MathJsChain<MathArray>): MathJsChain<MathArray>
+    abs(this: MathJsChain<Matrix>): MathJsChain<Matrix>
+    abs(this: MathJsChain<Unit>): MathJsChain<Unit>
 
     /**
      * Add two values, x + y. For matrices, the function is evaluated
      * element wise.
      * @param y Second value to add
      */
-    add(y: MathType): MathJsChain
+    add(this: MathJsChain<MathArray>, y: MathType): MathJsChain<MathArray>
+    add(this: MathJsChain<Matrix>, y: MathType): MathJsChain<Matrix>
+    add(this: MathJsChain<Unit>, y: MathType): MathJsChain<Unit>
+    add(
+      this: MathJsChain<MathNumericType>,
+      y: MathType
```

Comment:
If `y` is a Matrix or Array, the output will be a Matrix or Array (and not a MathNumericType)

Reply:
Changed it to behave the same as the normal `add` type for now for consistency